manifests/cert.pp

```diff
@@ -2,76 +2,149 @@
 
 define sslcert::cert (
   $certname,
-  $ensure   = 'present',
-  $csr      = '',
-  $certpath = $sslcert::certpath,
-  $chain    = '',
-  $owner    = 'root',
-  $group    = 'root',
-  $mode     = '0644',
-  $cert     = '',
-  $keypath  = $sslcert::keypath,
-  $keyname  = undef,
-  $keyowner = undef,
-  $keygroup = undef,
-  $keymode  = undef,
-  $key      = '',
+  $ensure              = 'present',
+  $csr                 = '', # not used, don't ask
+  $cert                = '', # actual cert
+  $certpath            = $::sslcert::certpath,# backward compatiblity (directory)
+  $certfile            = '', # complete path of file (dir+file)
+  $certowner           = undef, # default same as owner
+  $certgroup           = undef, # default same as group
+  $certmode            = undef, # default same as mode
+  $certtype            = '', # needed for digicert request
+  $chain               = '', # actual chain
+  $chainpath           = $::sslcert::certpath, #backward compatiblity (directory)
+  $chainfile           = '', # complete path of file (dir+file)
+  $chainowner          = undef, # default same as owner
+  $chaingroup          = undef, # default same as group
+  $chainmode           = undef, # default same as mode
+  $chainenabled        = $::sslcert::chainenabled,
```

Comment:
Can you explain why you added these extra parameters? In the definition we can already assign the legacy parameters.